lightning/src/ln/channelmanager.rs

```diff
@@ -2238,12 +2312,17 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 											htlcs.push(claimable_htlc);
 											for htlc in htlcs.iter() {
 												total_value += htlc.value;
-												if htlc.payment_data.total_msat != payment_data.total_msat {
-													log_trace!(self.logger, "Failing HTLCs with payment_hash {} as the HTLCs had inconsistent total values (eg {} and {})",
-														log_bytes!(payment_hash.0), payment_data.total_msat, htlc.payment_data.total_msat);
-													total_value = msgs::MAX_VALUE_MSAT;
+												match &htlc.claim_type {
+													ClaimableType::Invoice(pmt_data) => {
+														if pmt_data.total_msat != payment_data.total_msat {
+															log_trace!(self.logger, "Failing HTLCs with payment_hash {} as the HTLCs had inconsistent total values (eg {} and {})",
+																				 log_bytes!(payment_hash.0), payment_data.total_msat, pmt_data.total_msat);
+															total_value = msgs::MAX_VALUE_MSAT;
+														}
+														if total_value >= msgs::MAX_VALUE_MSAT { break; }
+													},
+													_ => unreachable!(),
```

Comment:
Are you sure this is unreachable? `htlc` here is an existing one, but I don't see where you've checked that its not just a pending-to-be-claimed keysend payment? Can you add a test where
* A sends B a part of an MPP payment and then a keysend payment
* A sends B a regular payment and then a keysend payment before B claims it
* A sends B a keysend payment and then a regular/MPP payment before B claims it